src/development/packages-and-plugins/plugin-testing.md

```diff
@@ -0,0 +1,140 @@
+---
+title: Plugins in Flutter tests
+short-title: Plugin tests
+description: Adding plugin as part of your Flutter tests.
+---
+
+Almost all [Flutter plugins][] have two parts:
+
+* Dart code, which provides the API your code calls.
+* Code written in a platform-specific (or "host") language,
+  such as Kotlin or Swift, which implements those APIs.
+
+In fact, the native (or host) language code distinguishes
+a plugin package from a standard package.
+
+[Flutter plugins]: {{site.url}}/development/packages-and-plugins/using-packages
+
+Building and registering the host portion of a plugin
+is part of the Flutter application build process,
+so plugins only work when your code is running
+in your application, such as with `flutter run`
+or when running [integration tests][].
+When running [Dart unit tests][] or
+[widget tests][], the host code isn’t available.
+If the code you are testing calls any plugins,
+this often results in errors like the following:
+
+```sh
+MissingPluginException(No implementation found for method someMethodName on channel some_channel_name)
+```
+
+[Dart unit tests]: {{site.url}}/cookbook/testing/unit/introduction
+[integration tests]: {{site.url}}/cookbook/testing/integration/introduction
+[widget tests]: {{site.api}}/flutter/flutter_test/flutter_test-library.html
+
+{{site.alert.note}}
+  Plugin implementations that [only use Dart][]
+  will work in unit tests. This is an implementation
+  detail of the plugin, however,
+  so tests shouldn't rely on it.
+{{site.alert.end}}
+
+[only use Dart]: {{site.url}}/development/packages-and-plugins/developing-packages
+
+When unit testing code that uses plugins,
+there are several options to avoid this exception.
+The following solutions are listed in order of preference.
+
+## Wrap the plugin
+
+In most cases, the best approach is to wrap plugin
+calls in your own API,
+and provide a way of [mocking][] that API in tests.
```

Comment:
sugestion: and provide a way of [mocking][] your own API in tests.

The current wording is ambiguous on first reading.

Reply:
done